frontend/src/__tests__/cypress/cypress/e2e/pipelines/PipelinesTopology.cy.ts

```diff
@@ -153,4 +210,71 @@ describe('Pipeline topology', () => {
+      pipelineRunDetails.findRunTaskRightDrawer().findByText('Logs').should('be.visible');
+      // test whether the logs load in Logs tab
+      pipelineRunDetails.findRightDrawer().findRightDrawerLogsTab().click();
+      pipelineRunDetails
+        .findRunTaskRightDrawer()
+        .findByTestId('logs-success-alert')
+        .should('be.visible');
+      pipelineRunDetails
+        .findRunTaskRightDrawer()
+        .findByTestId('logs')
+        .contains(
+          'sample log for namespace test-project, pod name conditional-execution-pipeline-0858f-flip-coin-pod and for step step-main',
+        );
+      // test whether single step logs download dropdown item is enabled when logs are available
+      pipelineRunDetails.findRunTaskRightDrawer().findByTestId('download-steps-toggle').click();
+      pipelineRunDetails
+        .findRunTaskRightDrawer()
+        .findByTestId('download-current-step-logs')
+        .should('not.be.disabled');
+      pipelineRunDetails.findRunTaskRightDrawer().findByTestId('download-steps-toggle').click();
+      pipelineRunDetails
+        .findRunTaskRightDrawer()
+        .findByTestId('download-current-step-logs')
+        .should('not.exist');
+      // test whether the raw logs dropddown item is enabled when logs are available
+      pipelineRunDetails.findRunTaskRightDrawer().findByTestId('logs-kebab-toggle').click();
+      pipelineRunDetails
+        .findRunTaskRightDrawer()
+        .findByTestId('raw-logs')
+        .should('not.be.disabled');
+      pipelineRunDetails.findRunTaskRightDrawer().findByTestId('logs-kebab-toggle').click();
+      pipelineRunDetails.findRunTaskRightDrawer().findByTestId('raw-logs').should('not.exist');
+      // test logs of another step
+      pipelineRunDetails.findStepSelect().should('not.be.disabled');
+      pipelineRunDetails.selectStepByName('step-copy-artifacts');
+      pipelineRunDetails
+        .findRunTaskRightDrawer()
+        .findByTestId('logs')
+        .contains('No logs available');
+      // test whether single step logs download dropdown item is disabled when logs are not available
+      pipelineRunDetails.findRunTaskRightDrawer().findByTestId('download-steps-toggle').click();
+      pipelineRunDetails.findRunTaskRightDrawer().findByTestId('download-current-step-logs');
+      pipelineRunDetails
+        .findRunTaskRightDrawer()
+        .findByTestId('download-current-step-logs')
+        .should('not.be.enabled');
+      pipelineRunDetails.findRunTaskRightDrawer().findByTestId('download-steps-toggle').click();
+      // test whether the raw logs dropddown item is disabled when logs are not available
+      pipelineRunDetails.findRunTaskRightDrawer().findByTestId('logs-kebab-toggle').click();
+      pipelineRunDetails.findRunTaskRightDrawer().findByTestId('raw-logs').should('not.be.enabled');
+      pipelineRunDetails.findRunTaskRightDrawer().findByTestId('logs-kebab-toggle').click();
```

Comment:
Why did you not extend the test object and have specific selectors for each test item you're trying to grab... you do several of these over and over and now have duplicate selectors instead of this simply saying:
```
pipelineRunLogs.findRawLogs().should('not.be.enabled')
```